CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+cmake_minimum_required(VERSION 3.14)
```

Comment:
for now, we should not include any build files that aren't just meant for test programs. This library will be built differently depending on the platform so we shouldn't be restricting the build tools the user wants to use, e.g. for a linux project we may use CMake, but for STM32 we're going to want to use the Makefile generated by CubeMX